include/deal.II/base/work_stream.h

```diff
@@ -874,14 +1019,352 @@ namespace WorkStream
+                      const Iterator &, ScratchData &, CopyData &)> &>(worker))
+                  worker(i, scratch_data, copy_data);
+                if (static_cast<const std::function<void(const CopyData &)> &>(
+                      copier))
+                  copier(copy_data);
+              }
+
+            return;
+          }
+
+        tf::Executor &executor = MultithreadInfo::get_taskflow_executor();
+        tf::Taskflow  taskflow;
+
+        tf::Task last_copier = taskflow.placeholder();
+
+        Threads::ThreadLocalStorage<std::unique_ptr<ScratchData>>
+          thread_local_scratch;
+
+        std::vector<std::unique_ptr<Chunk<CopyData>>> chunks;
+
+        unsigned int idx             = 0;
+        unsigned int remaining_items = std::distance(begin, end);
+
+        const unsigned int real_chunk_size =
+          (remaining_items / chunk_size < 3 * MultithreadInfo::n_threads()) ?
+            1 :
+            chunk_size;
+
+        Iterator it = begin;
+        while (it != end)
+          {
+            unsigned int count  = std::min(remaining_items, real_chunk_size);
+            Iterator     middle = it;
+            std::advance(middle, count);
+
+            chunks.emplace_back();
+
+            // this chunk works on [it,middle)
+            auto worker_task =
+              taskflow
+                .emplace([it_begin = it,
+                          it_end   = middle,
+                          idx,
+                          count,
+                          &sample_scratch_data,
+                          &thread_local_scratch,
+                          &chunks,
+                          &sample_copy_data,
+                          &worker]() {
+                  auto &scratch_ptr = thread_local_scratch.get();
+                  if (!scratch_ptr.get())
+                    scratch_ptr =
+                      std::make_unique<ScratchData>(sample_scratch_data);
+
+                  ScratchData &scratch = *scratch_ptr.get();
+                  chunks[idx] =
+                    std::make_unique<Chunk<CopyData>>(count, sample_copy_data);
+
+                  unsigned int counter = 0;
+                  for (Iterator it = it_begin; it != it_end; ++it, ++counter)
+                    {
+                      worker(it,
+                             scratch,
+                             chunks[idx].get()->copy_datas[counter]);
+                    }
+                })
+                .name("work");
+
+            tf::Task copier_task =
+              taskflow
+                .emplace([idx, &chunks, &copier]() mutable {
+                  auto chunk = chunks[idx].get();
+                  for (auto &cd : chunk->copy_datas)
+                    copier(cd);
+
+                  chunks[idx].reset();
+                })
+                .name("copy");
+
+            worker_task.precede(copier_task);
+
+            last_copier.precede(copier_task);
+            last_copier = copier_task;
```

Comment:
Ah, interesting -- tasks are copyable, so they are probably just a shim/pointer -- right?

Reply:
yes, it wasn't obvious from the documentation, but I saw it used like this in one of the examples.